Raven.Database/Impl/BackgroundTaskExecuter/RavenThreadPool.cs

```diff
@@ -468,8 +469,15 @@ private ThreadTask GetNextTask()
 
             WaitForBatchToCompletion(batchesCountdown);
 
-            if (exceptions.Count > 0)
-                throw new AggregateException(exceptions);
+            switch (exceptions.Count)
+            {
+                case 0:
+                    return;
+                case 1:
+                    throw exceptions.First();
```

Comment:
Bad, you are losing the original stack trace. Better to use this:
https://msdn.microsoft.com/en-us/library/system.runtime.exceptionservices.exceptiondispatchinfo.capture(v=vs.110).aspx

And then rethrow